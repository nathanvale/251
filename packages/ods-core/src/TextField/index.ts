export { TextField } from "./TextField";
export type TextFieldProps = import("./TextField").TextFieldProps;
