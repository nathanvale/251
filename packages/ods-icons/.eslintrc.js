module.exports = {
  overrides: [
    {
      files: ["src/mui/!(index)*.ts", "src/custom/!(index)*.tsx"],
      rules: {
        "import/newline-after-import": "off",
        "import/no-default-export": "off",
      },
    },
  ],
};
