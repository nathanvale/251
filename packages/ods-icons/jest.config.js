const base = require("../../jest.settings.js");
const pack = require("./package.json");

module.exports = {
  ...base,
  coveragePathIgnorePatterns: ["/ods-icons/src/custom/", "/ods-icons/src/mui/"],
  displayName: pack.name,
  name: pack.name,
  globals: {
    __DEV__: true,
  },
};
