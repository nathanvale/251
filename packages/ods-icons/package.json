{
  "private": false,
  "name": "@origin-digital/ods-icons",
  "description": "",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.6.4",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@origin-digital/ods-types": "^0.14.1"
  },
  "peerDependencies": {
    "react": "^16.9.0",
    "styled-components": "3.4.5"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
