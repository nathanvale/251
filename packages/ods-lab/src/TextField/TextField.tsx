import * as React from "react";
import MuiTextField, {
  StandardTextFieldProps as IStandardTextFieldProps,
} from "@material-ui/core/TextField";
import { InputAdornment } from "@origin-digital/ods-core";
import { Icon } from "..";

export interface TextFieldProps extends IStandardTextFieldProps {
  "data-id": string;
}

export const TextField = (props: TextFieldProps) => {
  return (
    <MuiTextField
      InputProps={{
        endAdornment: props.value && !props.disabled && (
          <InputAdornment position="end">
            <Icon color={props.error ? "error" : "action"} fontSize="small">
              {props.error ? "error" : "check_circle"}
            </Icon>
          </InputAdornment>
        ),
      }}
      {...props}
    />
  );
};
