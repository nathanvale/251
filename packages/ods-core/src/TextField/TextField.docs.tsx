import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { removeProps } from "@origin-digital/ods-helpers";
import { IconCheck } from "@origin-digital/ods-icons";
import { Stack } from "../Stack/Stack";
import { Text } from "../Text/Text";
import { Columns } from "../Columns/Columns";
import { Column } from "../Column/Column";
import { Box } from "../Box/Box";
import { textFieldBasePropDescriptions } from "../TextFieldBase/TextFieldBase.docs";
import { TextField, TextFieldProps } from "./TextField";

export const docs: ComponentDocs<TextFieldProps> = {
  category: "Interaction",
  componentName: "TextField",
  description: [
    "This is the underlying component of a number higher level Field components,",
    "such as TextField, SelectField, TextAreaField, etc.",
  ].join(" "),
  propDescriptions: {
    ...removeProps(textFieldBasePropDescriptions("TextField"), [
      "startAdornment",
      "endAdornment",
    ]),
    endIcon: [
      `Provide an icon to be shown at the right end of the TextField. If the value is string and one of`,
      `"success" | "error" | "validating", it will show the corresponding icon.`,
      `It also accepts any general icon as a react node, or in fact any element could be passed down to it,`,
      `e.g. an IconButton. If disabled={true} and endIcon is a string, the icon will not be shown. However,`,
      `custom icons will always be shown. It is the consumers responsibility to handle the visuals of the icon`,
      `for "disabled" and "error" states.`,
    ].join(" "),
    startIcon: [
      `Renders the received react element (mainly an icon) before (on the left of) the text in TextField`,
    ].join(" "),
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => <TextField id="text-field" label="Filled input" />,
    },
    additional: [
      {
        label: "Keep helperText space",
        description: [
          "You can set a flag to reserve the space that might be used by helperText.",
          "This allows us have a better user experience by not dynamically changing",
          "the height of the TextField which means the other components under it won't shift up and down.",
        ].join(" "),
        Code: () => (
          <Stack>
            <TextField
              id="text-field12"
              label="This keeps space for helperText even though it is empty"
              endIcon="success"
              reserveHelperTextSpace
            />
            <TextField
              id="text-field13"
              label="This does not take up space for helperText"
            />
            <TextField
              id="text-field14"
              label="Type here"
              helperText="This will be the error"
            />
          </Stack>
        ),
      },
      {
        label: "Success",
        description:
          "TextField with a green check icon, the TextField is in success state",
        Code: () => (
          <TextField
            id="text-field1"
            label="Type here"
            helperText="This will be the error"
            placeholder="No values here"
            endIcon="success"
          />
        ),
      },
      {
        label: "Error",
        description: "TextField is in error state",
        Code: () => (
          <TextField
            id="text-field1"
            label="Type here"
            helperText="This will be the error"
            placeholder="No values here"
            error
          />
        ),
      },
      {
        label: "Validating",
        description: "TextField is in validating state",
        Code: () => (
          <TextField
            id="text-field1"
            label="Type here"
            helperText="This will be the error"
            placeholder="No values here"
            endIcon="validating"
          />
        ),
      },
      {
        label: "Custom Icon at the end",
        description:
          "TextField accepts any custom icon to render after the text",
        Code: () => (
          <TextField
            id="text-field15"
            label="Type here"
            helperText="This will be the error"
            placeholder="No values here"
            endIcon={<IconCheck color="#a5bb48" />}
          />
        ),
      },
      {
        label: "Custom Icon at the start",
        description:
          "TextField accepts any custom icon to render before of the text",
        Code: () => (
          <TextField
            id="text-field16"
            label="Type here"
            helperText="This will be the error"
            placeholder="No values here"
            startIcon={<IconCheck color="#a5bb48" />}
          />
        ),
      },
      {
        label: "Default value",
        description: "One can set a default value for the TextField",
        Code: () => (
          <TextField
            id="text-field3"
            label="Name"
            helperText="Your name is required"
            defaultValue="John Smith"
          />
        ),
      },
      {
        label: "Handle change",
        description:
          "To handle change listen on onChange and provide the value prop",
        playroom: false,
        Code: () => {
          const [value, setValue] = React.useState<string>();
          return (
            <Stack space="medium">
              <TextField
                id="text-field4"
                label="Name"
                helperText="Enter your name please"
                value={value}
                onChange={(e) => setValue(e.target.value)}
              />
              {value && <Text tone="positive">Entered name: {value}</Text>}
            </Stack>
          );
        },
        codeString: `const [value, setValue] = React.useState<string>();
return (<Stack space="medium">
  <TextField
    id="text-field4"
    label="Name"
    helperText="Enter you name please"
    value={value}
    onChange={e => setValue(e.target.value)}
  />
  {value && <Text>Entered name: {value}</Text>}
</Stack>)`,
      },
      {
        label: "How to set width",
        description: [
          "TextField is always full width. So in order to set its width to a specific value",
          "it should be wrapped in a parent container which has its width set to the desired value.",
          "For most cases you should use Columns/Column as the parent to set the width to match",
          "our design system proportions. In case a pixel perfect width is needed wrap it with a",
          "Box component and set the width on the Box.",
        ].join(" "),
        Code: () => (
          <Stack space="medium">
            <Columns space="medium">
              <Column width="1/3">
                <TextField
                  id="text-field5"
                  label="Account No"
                  helperText="Column parent 1/3"
                />
              </Column>
              <Column width="2/3">
                <TextField
                  id="text-field6"
                  label="Account Name"
                  helperText="Column parent 2/3"
                />
              </Column>
            </Columns>
            <Box style={{ width: "8ch" }}>
              <TextField id="text-field7" label="BSB" helperText="Box parent" />
            </Box>
          </Stack>
        ),
      },
      {
        label: "size",
        description: "TextField could be compact taking less vertical space",
        Code: () => (
          <Columns space="medium">
            <Column>
              <TextField id="text-field8" label="Compact" size="small" />
            </Column>
            <Column>
              <TextField id="text-field9" label="Normal" />
            </Column>
          </Columns>
        ),
      },
      {
        label: "Outlined variant",
        description: ["TextField also has an Outlined variant."].join(" "),
        Code: () => (
          <TextField
            id="text-field11"
            label="Type here"
            variant="outlined"
            helperText="helper text for outlined input"
            placeholder="Placeholder"
          />
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Basic",
      Code: () => (
        <TextField
          id="text-field"
          label="label"
          placeholder="placeholder"
          helperText="details"
        />
      ),
    },
    {
      label: "Error",
      Code: () => (
        <TextField
          id="text-field"
          label="label"
          placeholder="placeholder"
          helperText="details"
          error={true}
        />
      ),
    },
    {
      label: "Success",
      Code: () => (
        <TextField
          id="text-field"
          label="label"
          placeholder="placeholder"
          helperText="details"
          endIcon="success"
        />
      ),
    },
    {
      label: "Validating",
      Code: () => (
        <TextField
          id="text-field"
          label="label"
          placeholder="placeholder"
          helperText="details"
          endIcon="validating"
        />
      ),
    },
    {
      label: "Custom endIcon",
      Code: () => (
        <TextField
          id="text-field"
          label="label"
          placeholder="placeholder"
          helperText="details"
          endIcon={<IconCheck color="#a5bb48" />}
        />
      ),
    },
    {
      label: "Custom startIcon",
      Code: () => (
        <TextField
          id="text-field"
          label="label"
          placeholder="placeholder"
          helperText="details"
          startIcon={<IconCheck color="#a5bb48" />}
        />
      ),
    },
    {
      label: "Set width",
      Code: () => (
        <Stack space="medium">
          <Box style={{ width: "250px" }}>
            <TextField id="text-field" label="BSB" helperText="6 digits" />
          </Box>
        </Stack>
      ),
    },
    {
      label: "Receive changes",
      Code: () => (
        <TextField
          id="text-field"
          label="label"
          placeholder="placeholder"
          helperText="details"
          onChange={(e) => {
            /* eslint-disable-next-line no-console */
            console.log("Change received: ", e.target.value);
          }}
        />
      ),
    },
  ],
};
