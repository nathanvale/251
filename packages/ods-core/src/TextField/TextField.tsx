import React from "react";
import { Theme, useTheme } from "@material-ui/core";
import { IconCheckCircle, IconError } from "@origin-digital/ods-icons";
import { BaseFieldProps, TextFieldBase } from "../TextFieldBase/TextFieldBase";
import { InputAdornment } from "../InputAdornment/InputAdornment";
import { Spinner } from "../Spinner/Spinner";
import {
  useFadeInAnimation,
  useBounceInAnimation,
} from "../_private/animations/animation-hooks";

export type EndIconType = "success" | "error" | "validating" | React.ReactNode;

export interface TextFieldProps extends BaseFieldProps {
  domProps?: React.HTMLAttributes<HTMLInputElement>;
  endIcon?: EndIconType;
  startIcon?: React.ReactNode;
}

export interface EndAdornmentProps {
  disabled?: boolean;
  error?: boolean;
  endIcon?: EndIconType;
  theme: Theme;
}

const TransitionContainer = ({
  children,
  className,
}: {
  children?: React.ReactNode;
  className?: string;
}) => (
  <InputAdornment className={className} position="end">
    {children}
  </InputAdornment>
);

const EndAdornment = ({
  disabled,
  error,
  endIcon,
  theme,
}: EndAdornmentProps) => {
  const fadeInClassName = useFadeInAnimation({
    duration: theme.transitions.duration.complex,
  });

  const bounceInClassName = useBounceInAnimation({
    duration: theme.transitions.duration.complex,
  });

  if (typeof endIcon !== "undefined" && typeof endIcon !== "string") {
    return <TransitionContainer>{endIcon}</TransitionContainer>;
  }
  if (disabled) {
    return null;
  }
  if (endIcon === "error" || (error && !endIcon)) {
    return (
      <TransitionContainer className={fadeInClassName}>
        <IconError tone="critical" />
      </TransitionContainer>
    );
  }
  if (endIcon === "success") {
    return (
      <TransitionContainer className={bounceInClassName}>
        <IconCheckCircle tone="positiveLight" />
      </TransitionContainer>
    );
  }

  if (endIcon === "validating") {
    return (
      <InputAdornment position="end">
        <Spinner />
      </InputAdornment>
    );
  }

  return null;
};

const StartAdornment = ({ startIcon }: { startIcon: React.ReactNode }) => {
  return <InputAdornment position="start">{startIcon}</InputAdornment>;
};
export const TextField = ({
  disabled,
  error,
  endIcon,
  startIcon,
  ...rest
}: TextFieldProps) => {
  const theme = useTheme();
  return (
    <TextFieldBase
      {...rest}
      endAdornment={
        (error || endIcon) && (
          <EndAdornment
            disabled={disabled}
            error={error}
            endIcon={endIcon}
            theme={theme}
          />
        )
      }
      startAdornment={startIcon && <StartAdornment startIcon={startIcon} />}
      disabled={disabled}
      error={error}
    />
  );
};

TextField.displayName = "TextField";
TextField.defaultProps = {
  variant: "filled",
  size: "medium",
};
