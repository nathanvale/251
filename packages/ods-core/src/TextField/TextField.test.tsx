import React from "react";
import {
  fireEvent,
  queryByAttribute,
  render,
} from "@origin-digital/ods-testing-library";
import { IconCheck } from "@origin-digital/ods-icons";
import { EndIconType, TextField } from "./TextField";

(["success", "validating", "error"] as EndIconType[]).forEach(
  (endIcon: EndIconType) => {
    describe(`TextField with ${endIcon} endIcon`, () => {
      test(`should render ${endIcon} after the text`, () => {
        const { container } = render(
          <TextField
            endIcon={endIcon}
            id="text-field"
            label={`TextField with ${endIcon} icon`}
          />
        );

        expect(container).toMatchSnapshot();
      });
    });
  }
);

test(`should render the custom after the text`, () => {
  const { container } = render(
    <TextField endIcon={<IconCheck />} id="text-field" label="Custom endIcon" />
  );

  expect(container).toMatchSnapshot();
});

test(`should render the custom before the text`, () => {
  const { container } = render(
    <TextField
      startIcon={<IconCheck />}
      id="text-field"
      label="Custom startIcon"
    />
  );

  expect(container).toMatchSnapshot();
});
test(`should render a TextField in error state`, () => {
  const { container } = render(
    <TextField
      id="my-text-field"
      label="The title"
      helperText="This is the info"
      error
    />
  );

  const label = queryByAttribute("data-id", container, "my-text-field-label");
  const helperText = queryByAttribute(
    "data-id",
    container,
    "my-text-field-helper-text"
  );

  expect(label).toHaveClass("Mui-error");
  expect(helperText).toHaveClass("Mui-error");
});

test("it passes down id and data-id", () => {
  const { getByTestId } = render(
    <TextField data-id="text-field-base" id="my-text-field" />
  );

  const TextFieldRoot = getByTestId("text-field-base") as HTMLDivElement;
  const input = getByTestId("text-field-base-input") as HTMLInputElement;

  expect(TextFieldRoot).toBeDefined();
  expect(input?.tagName.toLowerCase()).toEqual("input");
});

test("when a text is entered onChanged is called with the value of input", () => {
  const onChange = jest.fn();
  const { getByTestId } = render(
    <TextField id="my-text-field" onChange={onChange} />
  );

  const input = getByTestId("my-text-field-input") as HTMLInputElement;

  expect(input.value).toEqual("");

  fireEvent.change(input, { target: { value: "John" } });

  expect(onChange).toHaveBeenCalled();
  expect(input?.value).toEqual("John");
});
