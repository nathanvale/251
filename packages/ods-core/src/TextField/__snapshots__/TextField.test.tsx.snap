// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField with error endIcon should render error after the text 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-167 MuiFormControl-fullWidth"
    data-id="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-id="text-field-label"
      data-shrink="false"
      for="text-field"
      id="text-field-label"
    >
      TextField with error icon
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
    >
      <input
        aria-invalid="false"
        autocomplete="false"
        class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
        data-id="text-field-input"
        id="text-field"
        type="text"
        value=""
      />
      <span
        class="MuiInputAdornment-root makeStyles-fade-in-animation-228 MuiInputAdornment-filled MuiInputAdornment-positionEnd"
        data-id="input-adornment"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root svg-icon-critical-small-svg-icon-239"
          data-id="IconError"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TextField with success endIcon should render success after the text 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-1 MuiFormControl-fullWidth"
    data-id="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-id="text-field-label"
      data-shrink="false"
      for="text-field"
      id="text-field-label"
    >
      TextField with success icon
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
    >
      <input
        aria-invalid="false"
        autocomplete="false"
        class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
        data-id="text-field-input"
        id="text-field"
        type="text"
        value=""
      />
      <span
        class="MuiInputAdornment-root makeStyles-bounce-in-animation-64 MuiInputAdornment-filled MuiInputAdornment-positionEnd"
        data-id="input-adornment"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root svg-icon-positiveLight-small-svg-icon-73"
          data-id="IconCheckCircle"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TextField with validating endIcon should render validating after the text 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-83 MuiFormControl-fullWidth"
    data-id="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-id="text-field-label"
      data-shrink="false"
      for="text-field"
      id="text-field-label"
    >
      TextField with validating icon
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
    >
      <input
        aria-invalid="false"
        autocomplete="false"
        class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
        data-id="text-field-input"
        id="text-field"
        type="text"
        value=""
      />
      <span
        class="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionEnd"
        data-id="input-adornment"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
          data-id="spinner"
          role="progressbar"
          style="width: 20px; height: 20px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`should render the custom after the text 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-249 MuiFormControl-fullWidth"
    data-id="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-id="text-field-label"
      data-shrink="false"
      for="text-field"
      id="text-field-label"
    >
      Custom endIcon
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
    >
      <input
        aria-invalid="false"
        autocomplete="false"
        class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
        data-id="text-field-input"
        id="text-field"
        type="text"
        value=""
      />
      <span
        class="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionEnd"
        data-id="input-adornment"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root svg-icon-secondary-small-svg-icon-321"
          data-id="IconCheck"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`should render the custom before the text 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-331 MuiFormControl-fullWidth"
    data-id="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled"
      data-id="text-field-label"
      data-shrink="true"
      for="text-field"
      id="text-field-label"
    >
      Custom startIcon
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiFilledInput-adornedStart"
    >
      <span
        class="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionStart"
        data-id="input-adornment"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root svg-icon-secondary-small-svg-icon-399"
          data-id="IconCheck"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </span>
      <input
        aria-invalid="false"
        autocomplete="false"
        class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedStart MuiFilledInput-inputAdornedStart"
        data-id="text-field-input"
        id="text-field"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
